packages/jellyfish-api-core/__tests__/category/blockchain/getMempoolDescendants.test.ts

```diff
@@ -0,0 +1,111 @@
+  })
+
+  it('should return error if transaction is not in mempool', async () => {
+    const nonExistingTxId = 'a'.repeat(64)
+    const promise = testing.rpc.blockchain.getMempoolDescendants(nonExistingTxId)
+
+    await expect(promise)
+      .rejects
+      .toThrowError('RpcApiError: \'Transaction not in mempool\', code: -5, method: getmempooldescendants')
+  })
+
+  it('should return error if transaction id has an invalid length', async () => {
+    const invalidLengthTxid = 'a'.repeat(5)
+    const promise = testing.rpc.blockchain.getMempoolDescendants(invalidLengthTxid)
+
+    await expect(promise)
+      .rejects
+      .toThrow(`RpcApiError: 'parameter 1 must be of length 64 (not ${invalidLengthTxid.length}, for '${invalidLengthTxid}')', code: -8, method: getmempooldescendants`)
+  })
+})
+
+describe('Transactions with descendants', () => {
+  const container = new MasterNodeRegTestContainer()
+  const testing = Testing.create(container)
+
+  beforeAll(async () => {
+    await testing.container.start()
+    await testing.container.waitForWalletCoinbaseMaturity()
+  })
+
+  async function getTxIdWithDescendants (): Promise<string> {
+    const txId = await testing.rpc.wallet.sendToAddress('mwsZw8nF7pKxWH8eoKL9tPxTpaFkz7QeLU', 0.003)
+    for (let i = 0; i < 10; i++) {
+      await testing.rpc.wallet.sendToAddress('mwsZw8nF7pKxWH8eoKL9tPxTpaFkz7QeLU', 0.003)
+    }
+    return txId
+  }
+
+  afterAll(async () => {
+    await testing.container.stop()
+  })
+
+  it('should return JSON object if verbose is true', async () => {
+    const txIdWithDescendants = await getTxIdWithDescendants()
+    const mempoolDescendants = await testing.rpc.blockchain.getMempoolDescendants(txIdWithDescendants, true)
+
+    const keys = Object.keys(mempoolDescendants)
+    expect(keys.length).toBeGreaterThan(0)
+    for (const key of keys) {
+      expect(mempoolDescendants[key]).toStrictEqual({
+        fees: expect.any(Object),
+        vsize: expect.any(BigNumber),
+        weight: expect.any(BigNumber),
+        fee: expect.any(BigNumber),
+        modifiedfee: expect.any(BigNumber),
+        time: expect.any(BigNumber),
+        height: expect.any(BigNumber),
+        descendantcount: expect.any(BigNumber),
+        descendantsize: expect.any(BigNumber),
+        descendantfees: expect.any(BigNumber),
+        ancestorcount: expect.any(BigNumber),
+        ancestorsize: expect.any(BigNumber),
+        ancestorfees: expect.any(BigNumber),
+        wtxid: expect.any(String),
+        depends: expect.any(Array),
+        spentby: expect.any(Array),
+        'bip125-replaceable': expect.any(Boolean)
+      })
+    }
+  })
+
+  it('should return array of transaction ids if verbose is false', async () => {
+    const txIdWithDescendants = await getTxIdWithDescendants()
+    const mempoolDescendants = await testing.rpc.blockchain.getMempoolDescendants(txIdWithDescendants, false)
+    expect(mempoolDescendants.length).toBeGreaterThan(0)
```

Comment:
Wondering if there's any way to determine the specific length of descendants or does it change?

Reply:
They do change actually, even tho I create 10 descendant txes it returns only 3-4. Have a discussion [here](https://github.com/JellyfishSDK/jellyfish/pull/1784/files#r981851558) on this. Not sure how to write a deterministic test for this...

Reply:
since the number of descendants varies each time, would it be certain that `mempoolDescendants.length` will be greater than 0?

Reply:
That's a good point @delphk 🤔

Reply:
I think we can just stick to the current test, this test seems like not possible to make deterministic and the chances of descendants go `0` is very low with `10` following transactions, so I think its probably ok to check `toBeGreaterThan(0)` here. if the test starts to get flaky in future (unlikely) will think about it again!?

If not any opinions can someone help approve and merge this PR if no other issues present? Thanks! 🙏🏼